PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/StorageUploadWithoutLengthCheck.java

```diff
@@ -47,23 +54,55 @@ private static List<String> getMethodsToCheck() {
     @Override
     public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
         return new JavaRecursiveElementWalkingVisitor() {
+
             @Override
             public void visitMethodCallExpression(PsiMethodCallExpression expression) {
                 super.visitMethodCallExpression(expression);
                 String methodName = expression.getMethodExpression().getReferenceName();
 
                 if (METHODS_TO_CHECK.contains(methodName)) {
-                    PsiExpression[] arguments = expression.getArgumentList().getExpressions();
+                    boolean hasLengthArg = false;
+
+                    PsiExpressionList argumentList = expression.getArgumentList();
+                    PsiExpression[] arguments = argumentList.getExpressions();
+
+                    for (PsiExpression arg : arguments) {
+                        if (arg.getType().toString().equals("PsiType:long")) {
+                            hasLengthArg = true;
+                            break;
+                        }
+
+                        if (arg instanceof PsiMethodCallExpression) {
+                            PsiMethodCallExpression argMethodCall = (PsiMethodCallExpression) arg;
+                            PsiExpression qualifier = argMethodCall.getMethodExpression().getQualifierExpression();
 
-                    // TODO: check if a parameter object instantiated from a class with a length parameter field will NOT be detected.
-                    // eg line 150 stream parameter in sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/LargeBlobTests.java
-                    boolean hasLengthArg = Arrays.stream(arguments)
-                            .anyMatch(arg -> arg.getType().equalsToText("long"));
+                            while (qualifier instanceof PsiMethodCallExpression) {
```

Comment:
This long section goes through the method calls in a method call chain eg this.uploadWithResponse(new BlobParallelUploadOptions(data, length)
 .setParallelTransferOptions(parallelTransferOptions).setHeaders(headers).setMetadata(metadata).setTier(tier)
 .setRequestConditions(requestConditions), timeout, context);   It iterates from the last call to the first, then finally gets to PsiNewExpression to check for a length parameter

Reply:
Breakdown the large section into smaller sections of functional groupings and also include comments to explain the code.

Reply:
done!